#!/usr/bin/env python
PACKAGE = "fub_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("limit_steer_integral",          double_t, 0, "Max value for integral for steer control",                      0,    0,    100)
gen.add("low_speed_value",               double_t, 0, "Below this value [m/s] use low speed PID param set",            0.4,    0,     10)
gen.add("high_speed_value",              double_t, 0, "Above this value [m/s] use high speed PID param set",           1.0,      0,     10)
gen.add("kp_steer_low_speed_value",      double_t, 0, "P value for steer PID controller at low speed",                 1,      0,      1)
gen.add("kp_steer_high_speed_value",     double_t, 0, "P value for steer PID controller at high speed",                0.6,    0,      1)
gen.add("ki_steer_low_speed_value",      double_t, 0, "I value for steer PID controller at low speed",                 0,      0,      1)
gen.add("ki_steer_high_speed_value",     double_t, 0, "I value for steer PID controller at high speed",                0,      0,      1)
gen.add("kd_steer_low_speed_value",      double_t, 0, "D value for steer PID controller at low speed",                 0,      0,      10)
gen.add("kd_steer_high_speed_value",     double_t, 0, "D value for steer PID controller at high speed",                1,      0,      10)

exit(gen.generate(PACKAGE, "fub_controller", "ControllerMigSteer"))
